src/client/components/pages/Reservations/Reservations.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useEffect } from "react";
+import { getReservations } from "../../../data/reservations";
+import "./Reservations.css"
+
+const Reservations = (props) => {
+    const [reservations, setReservations] = useState([]);
+    const [inputState, setInputstate] = useState(false);
+    const [name, setName] = useState("");
+    const [phoneNumber, setPhoneNumber] = useState("");
+    const [numberOfGuests, setNumberOfGuests] = useState("");
+    const [email, setEmail] = useState("");
+    const [mealId, setMealId] = useState("");
+    const [date, setDate] = useState("");
+
+
+    useEffect(() => {
+        async function fetchData() {
+            const data = await getReservations();
+            setReservations(data)
+        }
+        fetchData()
+    }, [])
+
+    function newReservation() {
+        if (email !== "") {
+            setInputstate(true);
+            fetch("/api/reservations", {
+                method: "POST",
+                headers: { "Content-Type": "application/json" },
+                body: JSON.stringify({
+                    contact_name: name,
+                    contact_phonenumber: phoneNumber,
+                    contact_email: email,
+                    meal_id: mealId,
+                    number_of_guests: numberOfGuests,
+                    created_date: date,
+                }),
+            })
+                .catch((e) => {
+                    setError(e);
+                    alert("not valid details");
+                })
+                .finally(() => {
```

Comment:
`finally` will be called when the promise has been rejected or resolved. You'll want to use `then` here to only handle the happy path.